File: frontend/src/types/coach.ts
export type StudyPlanStatus = "active" | "completed" | "archived";

export interface StudyPlanTask {
  id: string;
  sequence: number;
  day_offset: number;
  title: string;
  description?: string | null;
  category: string;
  estimated_minutes: number;
  resource_path?: string | null;
  is_done: boolean;
  created_at: string;
  updated_at: string;
}

export interface StudyPlan {
  id: string;
  title: string;
  summary?: string | null;
  status: StudyPlanStatus | string;
  weeks: number;
  focus_areas?: string[] | null;
  model_provider?: string | null;
  created_at: string;
  updated_at: string;
  task_count: number;
  done_count: number;
}

export interface StudyPlanDetail extends StudyPlan {
  tasks: StudyPlanTask[];
}

export interface CoachMessage {
  id: string;
  role: "user" | "assistant" | string;
  content: string;
  extra?: Record<string, unknown> | null;
  created_at: string;
}

export interface CoachAskResult {
  reply: CoachMessage;
  history: CoachMessage[];
}

export interface CoachInsight {
  headline: string;
  tips: string[];
  weak_topics: string[];
  focus_areas: string[];
  suggested_weeks: number;
}

File: backend/app/schemas/coach.py
"""AI coach and study plan schemas."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class GenerateStudyPlanRequest(BaseModel):
    weeks: int = Field(default=2, ge=1, le=8)
    title: str | None = Field(default=None, max_length=255)
    focus_areas: list[str] | None = Field(default=None, max_length=12)


class UpdateTaskRequest(BaseModel):
    is_done: bool


class CoachAskRequest(BaseModel):
    message: str = Field(min_length=1, max_length=2000)


class StudyPlanTaskResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    sequence: int
    day_offset: int
    title: str
    description: str | None = None
    category: str
    estimated_minutes: int
    resource_path: str | None = None
    is_done: bool
    created_at: datetime
    updated_at: datetime


class StudyPlanResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    title: str
    summary: str | None = None
    status: str
    weeks: int
    focus_areas: list | None = None
    model_provider: str | None = None
    created_at: datetime
    updated_at: datetime
    task_count: int = 0
    done_count: int = 0


class StudyPlanDetailResponse(StudyPlanResponse):
    tasks: list[StudyPlanTaskResponse] = Field(default_factory=list)


class CoachMessageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    role: str
    content: str
    extra: dict | None = None
    created_at: datetime


class CoachAskResponse(BaseModel):
    reply: CoachMessageResponse
    history: list[CoachMessageResponse]


class CoachInsightResponse(BaseModel):
    headline: str
    tips: list[str]
    weak_topics: list[str]
    focus_areas: list[str]
    suggested_weeks: int
